Makefile

```diff
@@ -23,77 +23,78 @@ pre-commit: ## Runs the pre-commit checks over entire repo
+	@ cd terraform/envs/$(env) && \
+	terraform init -backend-config='bucket=${VERTEX_PROJECT_ID}-tfstate' && \
+	terraform apply -var 'project_id=${VERTEX_PROJECT_ID}' -var 'region=${VERTEX_LOCATION}'
+
+undeploy: ## DESTROY the Terraform infrastructure in your project. Requires VERTEX_PROJECT_ID and VERTEX_LOCATION env variables to be set in env.sh. Optionally specify env=<dev|test|prod> (default = dev)
+	@ cd terraform/envs/$(env) && \
+	terraform init -backend-config='bucket=${VERTEX_PROJECT_ID}-tfstate' && \
+	terraform destroy -var 'project_id=${VERTEX_PROJECT_ID}' -var 'region=${VERTEX_LOCATION}'
 
-test-trigger: ## Runs unit tests for the pipeline trigger code
+install: ## Set up local environment for Python development on pipelines
 	@cd pipelines && \
-	poetry run python -m pytest tests/trigger
+	poetry install --with dev && \
+	cd .. && \
+	for component_group in components/*/ ; do \
+		echo "Setup for $$component_group" && \
+		cd "$$component_group" && \
+		poetry install --with dev && \
+		cd ../.. ;\
+	done ; \
 
-compile-pipeline: ## Compile the pipeline to pipeline.yaml. Must specify pipeline=<training|prediction>
+
+compile: ## Compile the pipeline to pipeline.yaml. Must specify pipeline=<training|prediction>
 	@cd pipelines/src && \
 	poetry run kfp dsl compile --py pipelines/${pipeline}/pipeline.py --output pipelines/${pipeline}/pipeline.yaml --function pipeline
 
-setup-components: ## Run unit tests for a component group
-	@cd "components/${GROUP}" && \
-	poetry install --with dev
-
-setup-all-components: ## Run unit tests for all pipeline components
-	@set -e && \
-	for component_group in components/*/ ; do \
-		echo "Setup components under $$component_group" && \
-		$(MAKE) setup-components GROUP=$$(basename $$component_group) ; \
-	done
-
-test-components: ## Run unit tests for a component group
-	@cd "components/${GROUP}" && \
-	poetry run pytest
-
-test-all-components: ## Run unit tests for all pipeline components
-	@set -e && \
-	for component_group in components/*/ ; do \
-		echo "Test components under $$component_group" && \
-		$(MAKE) test-components GROUP=$$(basename $$component_group) ; \
-	done
+targets ?= training serving
+build: ## Build and push training/serving container image using Docker. Specify target=<training|serving>
+	@cd model && \
+	for target in $$targets ; do \
+		echo "Building $$target image" && \
+		gcloud builds submit . \
+		--region=${VERTEX_LOCATION} \
+		--project=${VERTEX_PROJECT_ID} \
+		--gcs-source-staging-dir=gs://${VERTEX_PROJECT_ID}-staging/source \
+		--substitutions=_DOCKER_TARGET=${target},_DESTINATION_IMAGE_URI=${CONTAINER_IMAGE_REGISTRY}/${target}:${RESOURCE_SUFFIX} ; \
+	done 
+
+
+compile ?=true
+build ?= true 
+run: ## Compile pipeline and run pipeline in sandbox environment. Must specify pipeline=<training|prediction>. Optionally specify enable_pipeline_caching=<true|false> (defaults to default Vertex caching behaviour)
+	@if [ "${compile}" ]; then \
```

Comment:
also document how to use compile and build args

Reply:
5fb0503